code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -96,11 +130,319 @@
 		ticker.mode.update_necro_icons_added(Controller)
 		ticker.mode.update_all_necro_icons()
 		ticker.mode.risen.Add(Controller)
-	name += " ([rand(1,1000)])"
 
-/mob/living/simple_animal/hostile/necro/copy/ListTargets()
-	. = ..()
-	return . - creator
+	if(name == initial(name) && !unique_name)
+		name += " ([rand(1,1000)])"
+
+/mob/living/simple_animal/hostile/necro/zombie/Life()
+	..()
+	/*TODO
+	First, check if the zombie can potentially evolve
+	Have the zombie move to a corpse and start chewing at it
+	If neither of these things are applicable, break some lights to set the mood
+	Otherwise, start wandering and bust down some doors to find more food
+	*/
+	if(!isUnconscious())
+		if(stance == HOSTILE_STANCE_IDLE && (!client)) //Not doing anything at the time
+			var/list/can_see = view(src, vision_range)
+			if(!busy)
+				if(can_evolve)//Can we evolve, and have we fed
+					busy = EVOLVING
+					check_evolve()
+					busy = 0
+				if((health < maxHealth) || (maxHealth < health_cap) && !busy)
+					var/mob/living/carbon/human/C = find_food(can_see)//Is there something to eat in range?
+					if(C) //If so, chow down
+						Goto(C, speed)
+						busy = MOVING_TO_TARGET
+						give_up(C) //If we're not there in 10 seconds, give up
+						if(C.Adjacent(src) && busy != EATING) //Once we've finally caught up
+							busy = EATING
+							eat(C)
+							C = null
+							walk(src, 0)
+				if(!busy && break_doors != CANT)//So we don't try to eat and open doors
```

Comment:
Because that's what defines are for, you can edit the value just once instead of loads of times and it's enforced to be consistent between all of them.